### DISCUSSION TITLE
SEB 3.5.0 Win 10/11 - SEB LOCKED after restart

### DISCUSSION THREAD

**Discussion by kstarzecki** [SafeExamBrowser/seb-win-refactoring#652] [Q&A] [answered]
Hey, we're in a bit of a pickle. Hoping you could help us coming up with a solution.

We're running SEB in a controlled environment. Students come to the computer room with computers already logged in to 'exam user' and SEB launched. After the exam session is completed, computers are restarted/shutdown on schedule.

We're getting a red "SEB LOCKED" screen after a scheduled restart via shutdown.exe.
Any way to programmatically end the SEB session?

Thanks,
Chris

**Comment by danschlet** [comment]
A secure exam session in SEB (one with a quit password set) always has to be quit properly, e.g. using a Quit Link ([see manual](https://safeexambrowser.org/windows/win_usermanual_en.html#ExamPane)), by manually using the quit button and entering the quit password or with SEB Server (there you can quit single or all clients connected to an exam).

A proper SEB integration in an assessment system should ideally automatically generate SEB settings containing the quitURL setting and display a quit link/button with the same URL after the exam was submitted.

If you're implementing your own assessment system, you could integrate a feature which invokes a redirect to the quitURL defined in your SEB settings. Then SEB would be correctly quit and you could restart or shutdown your clients, without the SEB Locked screen being displayed if you start another session with the same SEB settings later.

**Comment by kstarzecki** [comment]
Thing is, it didn't give us any problems up to 3.5.0. We're not using SEB Server, and relying on users to terminate the session is not reliable. 

Any other options? What registry keys, files are created/changed in the lockdown process? We could restore those ourselves if we had the list. Having resetUtility that could be automated (for current user) would be a great addition, since we could add a policy to run it before shutdown.

**Comment by danschlet** [comment]
Other SEB users insisted on this new feature in SEB 3.5 (SEB for macOS and iOS had this lock screen for not terminated sessions since long time), as it is a deterrence for students to for example force restart their BYOD machine during an exam.

You don't have to rely on users to terminate a session. I guess your assessment system (what are you using?) relies on students to submit a completed exam? Then simply use the URL of the page which is displayed after submitting the exam results as quitURL.

**Comment by kstarzecki** [comment]
We're not using one single system. We're using Canvas LMS, and teachers choose to use either classic or new canvas quiz, or one of the other platforms that we support (i.e. Möbius Courseware, Matlab Grader). Some have other forms of examination like essays, document scans, etc. This makes near impossible to have a quit url.

We have thought about adding a 'button' on the taskbar that just says 'end exam' which would contain the quitURL but that's not the best solution.

**Comment by kstarzecki** [comment]
Ideally, we'd like to implement "cleanup before start" and/or a "cleanup at stop" script/policy.
Knowing what to clean would make our job much easier.

**Reply by danschlet** [reply, marked as answer]
You can contact us by email info at safeexambrowser dot org.

**Reply by kstarzecki** [reply]
Gotcha!

**Comment by r3df0chs** [comment]
Hi @kstarzecki ,
did you find an solution for you problem? I also would like to reboot the clients through a Scheduled Task for example.

**Reply by kstarzecki** [reply]
We have hit some delays due to personnel shortages, and we're only now getting on with the workflows.
Once we have some feature requests, I'll add it to this thread. :)
